src/index.js

```diff
@@ -0,0 +1,144 @@
+// Literacy
+// ========
+// Literate programming in JavaScript using reStructuredText.
+//
+//
+// Register ``.js.rst``` File Extension
+// ------------------------------------
+// Intercept Node module loading for filenames ending in ``.js.rst``.  This
+// allows ``.js.rst`` files to be passed to ``require`` calls without including
+// the suffix.
+//
+// Implementation is copied from `register.coffee`_ in CoffeeScript and is similar
+// to `babel-register`_.
+//
+// .. _register.coffee: https://github.com/jashkenas/coffeescript/tree/master/src/register.coffee
+// .. _babel-register: https://github.com/babel/babel/tree/master/packages/babel-register
+//
+// The implementation uses the ``require.extensions`` global in Node. Setting a
+// key in this dictionary adds Node handling for new file extensions. The key is
+// the file suffix string to support, the value is a function to call to load and
+// process the file.
+//
+// But ``require.extenions`` has been deprecated in Node because there is the
+// potential for conflicts when a suffix is used by two modules simultaneously.
+// It also encourages the offloading of compilation to runtime instead of build
+// time.
+//
+// Strictly speaking it should not be used. However:
+//
+// - Only this module uses ``.js.rst`` suffixes.
+// - The Node module containing the ``require.extensions`` implementation is
+//   locked so Node is committed to keeping the current form indefinitely.
+// - Production use cases are encouraged to precompile ``.js.rst`` sources.
```

Comment:
I'm okay with this reasoning.

![](https://media.tenor.com/images/058ea6adef5d9f3ac298fee50f61d8ba/tenor.gif)

Reply:
I'm working on a bin script to make the precompilation feasible today. This require hook business will be put in a separate `literacy/register` so it will have to be explicitly included. My intended use case is a Webpack loader anyway so the precompilation will happen there really. This is all following the CoffeeScript model.

Good doggies.